standard-integration/server.js

```diff
@@ -1,34 +1,149 @@
-import "dotenv/config"; // loads variables from .env file
-import express from "express";
-import * as paypal from "./paypal-api.js";
-const {PORT = 8888} = process.env;
+import express from 'express';
+import fetch from 'node-fetch';
+import 'dotenv/config';
+import path from 'path';
 
+const { PAYPAL_CLIENT_ID, PAYPAL_CLIENT_SECRET } = process.env;
+const base = 'https://api-m.sandbox.paypal.com';
 const app = express();
 
-app.use(express.static("public"));
-
 // parse post params sent in body in json format
 app.use(express.json());
 
-app.post("/my-server/create-paypal-order", async (req, res) => {
+/**
+ * Generate an OAuth 2.0 access token for authenticating with PayPal REST APIs.
+ * @see https://developer.paypal.com/api/rest/authentication/
+ */
+const generateAccessToken = async () => {
+  try {
+    if (!PAYPAL_CLIENT_ID || !PAYPAL_CLIENT_SECRET) {
+      throw new Error('MISSING_API_CREDENTIALS');
+    }
+    const auth = Buffer.from(
+      PAYPAL_CLIENT_ID + ':' + PAYPAL_CLIENT_SECRET,
+    ).toString('base64');
+    const response = await fetch(`${base}/v1/oauth2/token`, {
+      method: 'POST',
+      body: 'grant_type=client_credentials',
+      headers: {
+        Authorization: `Basic ${auth}`,
+      },
+    });
+
+    const data = await response.json();
+    return data.access_token;
+  } catch (error) {
+    console.error('Failed to generate Access Token:', error);
+  }
+};
+
+/**
+ * Create an order to start the transaction.
+ * @see https://developer.paypal.com/docs/api/orders/v2/#orders_create
+ */
+const createOrder = async (cart) => {
+  // use the cart information passed from the front-end to calculate the purchase unit details
+  console.log(
+    'shopping cart information passed from the frontend createOrder() callback:',
+    cart,
+  );
+
+  const accessToken = await generateAccessToken();
+  const url = `${base}/v2/checkout/orders`;
+  const payload = {
+    intent: 'CAPTURE',
+    purchase_units: [
+      {
+        amount: {
+          currency_code: 'USD',
+          value: '0.02',
+        },
+      },
+    ],
+  };
+
+  const response = await fetch(url, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${accessToken}`,
+      // Uncomment one of these to force an error for negative testing (in sandbox mode only). Documentation:
+      // https://developer.paypal.com/tools/sandbox/negative-testing/request-headers/
+      // "PayPal-Mock-Response": '{"mock_application_codes": "MISSING_REQUIRED_PARAMETER"}'
+      // "PayPal-Mock-Response": '{"mock_application_codes": "PERMISSION_DENIED"}'
+      // "PayPal-Mock-Response": '{"mock_application_codes": "INTERNAL_SERVER_ERROR"}'
```

Comment:
👏 👏 👏

Reply:
These are great! I think it'd also be nice to add to a section to https://developer.paypal.com/sdk/js/ with something like `Troubleshooting` and these would show: "negative testing is done via the following headers"

Reply:
Great point @westeezy. Let's figure out how to best add content about these headers to the sdk docs.